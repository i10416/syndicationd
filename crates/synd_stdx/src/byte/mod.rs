pub type Byte = byte_unit::Byte;

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn parse() {
        assert_eq!(4_u64 * 1024 * 1024, Byte::parse_str("4MiB", true).unwrap());
        assert_eq!(4_u64 * 1000 * 1000, Byte::parse_str("4MB", true).unwrap());
    }
}
