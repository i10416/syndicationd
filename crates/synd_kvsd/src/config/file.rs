use std::{net::IpAddr, time::Duration};

use serde::Deserialize;
use synd_stdx::byte::Byte;

#[derive(Debug, Deserialize)]
pub(crate) struct ConfigFile {
    connection: Option<ConnectionEntry>,
    authentication: Option<AuthenticationEntry>,
    bind: Option<BindEntry>,
    tls: Option<TlsEntry>,
}

#[derive(Debug, Deserialize)]
pub(crate) struct ConnectionEntry {
    limit: Option<u32>,
    buffer_size: Option<Byte>,
}

#[derive(Debug, Deserialize)]
pub(crate) struct AuthenticationEntry {
    #[serde(
        default,
        deserialize_with = "synd_stdx::time::humantime::de::parse_duration_opt"
    )]
    timeout: Option<Duration>,
}

#[derive(Debug, Deserialize)]
pub(crate) struct BindEntry {
    address: Option<IpAddr>,
    port: Option<u16>,
}

#[derive(Debug, Deserialize)]
pub(crate) struct TlsEntry {
    disable: Option<bool>,
}

#[cfg(test)]
mod tests {
    use super::*;

    const CONFIG: &str = r#"
[connection]        
limit = 2048
buffer_size = "4MiB"

[authentication]
timeout = "3s"

[bind]
address = "127.0.0.1"
port = 7777

[tls]
disable = true

"#;

    #[test]
    fn deserialize() {
        let c: ConfigFile = toml::from_str(CONFIG).unwrap();
        insta::assert_debug_snapshot!("deserialized_config_file", c);
    }
}
