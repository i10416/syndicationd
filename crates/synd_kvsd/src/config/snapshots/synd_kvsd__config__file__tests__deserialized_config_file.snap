---
source: crates/synd_kvsd/src/config/file.rs
expression: c
---
ConfigFile {
    connection: Some(
        ConnectionEntry {
            limit: Some(
                2048,
            ),
            buffer_size: Some(
                Byte(
                    4194304,
                ),
            ),
        },
    ),
    authentication: Some(
        AuthenticationEntry {
            timeout: Some(
                3s,
            ),
        },
    ),
    bind: Some(
        BindEntry {
            address: Some(
                127.0.0.1,
            ),
            port: Some(
                7777,
            ),
        },
    ),
    tls: Some(
        TlsEntry {
            disable: Some(
                true,
            ),
        },
    ),
}
